stakingModel/contracts/UBIScheme.sol

```diff
@@ -266,29 +300,42 @@ contract UBIScheme is AbstractUBI {
     {
         require(accounts.length < gasleft().div(iterationGasLimit), "exceeds of gas limitations");
         for(uint256 i = 0; i < accounts.length; ++i) {
-            fish(accounts[i]);
+            if(isRegistered(accounts[i]) && !isActiveUser(accounts[i]) && !fishedUsersAddresses[accounts[i]]) {
+                fish(accounts[i]);
+            }
         }
         return true;
     }
 
     /* @dev Function for automate claiming UBI for users. Emits the caller address and the
      * given list length and the actual number of claimers.
-     * @param accounts - claimers account list
+     * @param _start - start index of the pending list
+     * @param _end - end index of the pending list
      * @return A bool indicating if UBI was claimed
      */
-    function distribute(address[] memory accounts)
+    function distribute(uint256 _start, uint256 _end)
         public
         requireActive
         returns (bool)
     {
-        require(accounts.length < gasleft().div(iterationGasLimit), "exceeds of gas limitations");
+        require(_end.sub(_start).add(1) < gasleft().div(iterationGasLimit), "exceeds of gas limitations");
         uint256 claimers = 0;
-        for(uint256 i = 0; i < accounts.length; ++i) {
-            if(identity.isWhitelisted(accounts[i]) && _claim(accounts[i])) {
-                    claimers = claimers.add(1);
+        setDay();
+        for(uint256 i = _start; i <= _end && i < pendingUserAddresses.length; ++i) {
+            address account = pendingUserAddresses[i];
+            uint256 newDistribution = distributionFormula(0, account);
+            // pending user might claims tokens twice in the first day
+            if(identity.isWhitelisted(account) &&
+            pendingUserClaims[account].isEligible &&
+            pendingUserClaims[account].day == currentDay) {
+                PendingUser memory user = pendingUserClaims[account];
+                user.isEligible = false;
+                pendingUserClaims[account] = user;
```

Comment:
why not use storage instead of memory?

Reply:
the same result over here. changed c7586741b9c670251e051b652b23e1d9ab534969